### ISSUE TITLE
Bugsnag doesn't send crash on RCTFatal() call 

### ISSUE THREAD

**Issue by RishatShamsutdinov** [bugsnag/bugsnag-js#1712]
https://github.com/bugsnag/bugsnag-js/blob/59e4de95790d72ae3a755d8201951f0de30bc6ca/packages/react-native/ios/BugsnagReactNative/BugsnagReactNativePlugin.m#L19

How to reproduce:

```Objective-C
[<RCT bridge> enqueueJSCall:<JS Module Name> method:<JS Method Name> args:<JS Method Args> completion:^{
    @throw [NSException exceptionWithName:@"MyException" reason:@"Manual" userInfo:nil];
}];
```

Exception will be thrown from https://github.com/facebook/react-native/blob/6031e4ab626b7aa57e6ed1bbf266e39220d61adc/React/Base/RCTAssert.m#L146.

I think that the filter in the subject doesn't work properly because in that case JS Exception will not be thrown but the exception will have `RCTFatalException` name.

P.S.: we faced the issue in our project.

A crash I found in device logs from Xcode:

```
Last Exception Backtrace:
0   CoreFoundation                         0x1a30d1d78 __exceptionPreprocess + 220
1   libobjc.A.dylib                        0x1bbd36734 objc_exception_throw + 60
2   MyApp                                  0x103c8917c RCTGetFatalHandler + 0
3   MyApp                                  0x103ca0d14 -[RCTCxxBridge handleError:] + 1232148 (RCTCxxBridge.mm:1211)
4   MyApp                                  0x103c9f250 __34-[RCTCxxBridge _initializeBridge:]_block_invoke + 1225296 (RCTCxxBridge.mm:719)
5   MyApp                                  0x103cb4724 facebook::react::RCTMessageThread::tryFunc(std::__1::function<void ()> const&) + 56
6   MyApp                                  0x103cb44b8 invocation function for block in facebook::react::RCTMessageThread::runAsync(std::__1::function<void ()>) + 1311928 (RCTMessageThread.mm:45)
7   CoreFoundation                         0x1a30ab0e4 __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 28
8   CoreFoundation                         0x1a30ac040 __CFRunLoopDoBlocks + 412
9   CoreFoundation                         0x1a3043ec8 __CFRunLoopRun + 1720
10  CoreFoundation                         0x1a30576b8 CFRunLoopRunSpecific + 600
11  MyApp                                  0x103c9c9b4 +[RCTCxxBridge runRunLoop] + 1214900 (RCTCxxBridge.mm:367)
12  Foundation                             0x1a488e55c __NSThread__start__ + 808
13  libsystem_pthread.dylib                0x21405a9ac _pthread_start + 148
14  libsystem_pthread.dylib                0x214059e68 thread_start + 8
```

**Comment by RishatShamsutdinov** [comment]
RN wraps exception to own one.

**Comment by luke-belton** [comment]
Hi @RishatShamsutdinov - [a fix](https://github.com/bugsnag/bugsnag-js/pull/1719) for this was released in [v7.16.4 ](https://github.com/bugsnag/bugsnag-js/releases/tag/v7.16.4)of bugsnag-js 🎉  Thanks again for reporting this!
